### PR TITLE
TEE Latency benchmark

### PR DESCRIPTION
**Pull request by igoropaniuk**
Add support for CFG_TEE_BENCHMARK and CONFIG_OPTEE_BENCHMARK options

Relates to: 
https://github.com/linaro-swg/linux/pull/27
https://github.com/OP-TEE/build/pull/81
https://github.com/OP-TEE/optee_test/pull/145
https://github.com/OP-TEE/optee_os/pull/1175
https://github.com/OP-TEE/optee_client/pull/63

### CODE REVIEW

File: common.mk [hunk 1]
```diff
@@ -14,11 +14,12 @@ OPTEE_OS_PATH			?= $(ROOT)/optee_os
 OPTEE_CLIENT_PATH		?= $(ROOT)/optee_client
 OPTEE_CLIENT_EXPORT		?= $(OPTEE_CLIENT_PATH)/out/export
 OPTEE_TEST_PATH			?= $(ROOT)/optee_test
-OPTEE_TEST_OUT_PATH 		?= $(ROOT)/optee_test/out
+OPTEE_TEST_OUT_PATH		?= $(ROOT)/optee_test/out
 HELLOWORLD_PATH			?= $(ROOT)/hello_world
 
 # default high verbosity. slow uarts shall specify lower if prefered
 CFG_TEE_CORE_LOG_LEVEL		?= 3
+CFG_TEE_BENCHMARK		?= y
```
Comment (jbech-linaro) [inline common.mk]: Do we really want to have this enabled by default? To me this feels like something you enable just when working with it.

File: kconfigs/qemu.conf [hunk 2]
```diff
@@ -1,5 +1,6 @@
 CONFIG_TEE=y
 CONFIG_OPTEE=y
+CONFIG_OPTEE_BENCHMARK=y
```
Comment (jbech-linaro) [inline kconfigs/qemu.conf]: Same comment here, a manual Kconfig change is what I'd prefer ... alternatively, you create a another *.conf file, that you merge with qemu.conf if `CFG_TEE_BENCHMARK=y`, something like [this](https://github.com/OP-TEE/build/blob/ba038a5e498fd2587ef1418842f06a598323663c/juno.mk#L73-L80).

File: common.mk (hunk 1 above)
Comment (igoropaniuk) [inline common.mk]: Sure, I just left it switched on just for this RFC PR
I'm going to set CFG_TEE_BENCHMARK to n by default in the final PR

File: common.mk [hunk 3]
```diff
@@ -14,11 +14,14 @@ OPTEE_OS_PATH			?= $(ROOT)/optee_os
...
 OPTEE_TEST_PATH			?= $(ROOT)/optee_test
-OPTEE_TEST_OUT_PATH 		?= $(ROOT)/optee_test/out
+OPTEE_TEST_OUT_PATH		?= $(ROOT)/optee_test/out
 HELLOWORLD_PATH			?= $(ROOT)/hello_world
 
 # default high verbosity. slow uarts shall specify lower if prefered
 CFG_TEE_CORE_LOG_LEVEL		?= 3
+CFG_TEE_BENCHMARK		?= n
+# print latency statistics for each TEEC_InvokeCommand
+CFG_TEE_PRINT_LATENCY_STAT ?= n
```
Comment (etienne-lms) [inline common.mk]: minor: align `?= n` with a tabulation (as in above line)

File: common.mk [hunk 4]
```diff
@@ -14,11 +14,14 @@ OPTEE_OS_PATH			?= $(ROOT)/optee_os
 OPTEE_CLIENT_PATH		?= $(ROOT)/optee_client
 OPTEE_CLIENT_EXPORT		?= $(OPTEE_CLIENT_PATH)/out/export
 OPTEE_TEST_PATH			?= $(ROOT)/optee_test
-OPTEE_TEST_OUT_PATH 		?= $(ROOT)/optee_test/out
+OPTEE_TEST_OUT_PATH		?= $(ROOT)/optee_test/out
 HELLOWORLD_PATH			?= $(ROOT)/hello_world
 
 # default high verbosity. slow uarts shall specify lower if prefered
 CFG_TEE_CORE_LOG_LEVEL		?= 3
+CFG_TEE_BENCHMARK		?= n
```
Comment (etienne-lms) [inline common.mk]: minor: add a common comment for both.

**Pull request closed without merging**